sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -0,0 +1,136 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/freestanding/limits.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+static const uint32_t kAonTimerFreq = 200 * 1000;  // 200kHz
+static const uint32_t kSleepTime_us = 4 * 1000;    // 2ms
+static dif_aon_timer_t aon_timer;
+static dif_rv_plic_t plic;
+static dif_aon_timer_irq_t irq;
+static top_earlgrey_plic_peripheral_t peripheral;
+
+static ibex_timeout_t timeout;
+uint64_t time_elapsed = 0;
+
+static void aon_timer_wakeup_config(dif_aon_timer_t *aon_timer,
+                                    uint32_t wakeup_threshold) {
+  // Make sure that wake-up timer is stopped.
+  CHECK_DIF_OK(dif_aon_timer_wakeup_stop(aon_timer));
+
+  // Make sure the wake-up IRQ is cleared to avoid false positive.
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge(aon_timer,
+                                             kDifAonTimerIrqWkupTimerExpired));
+
+  // Check whether the IRQ is pending
+  bool is_pending = true;
+  CHECK_DIF_OK(dif_aon_timer_irq_is_pending(
+      aon_timer, kDifAonTimerIrqWkupTimerExpired, &is_pending));
+  CHECK(!is_pending);
+
+  // Start timer
+  CHECK_DIF_OK(dif_aon_timer_wakeup_start(aon_timer, wakeup_threshold, 0));
```

Comment:
I think `aon_timer_wakeup_config` as the name is slightly misleading, when you have `start` function inside.

Reply:
I agree but now I`m using `aon_timer_testutils_wakeup_config` and rename it means change several tests. What about do it in another PR?

Reply:
A follow-up definitely seems like a good way to go. Even then, it's not really that important so feel free to ignore it.